bin/go-rpm-integration

```diff
@@ -155,45 +154,49 @@ while [ $# -gt 0 ] ; do
 done
 
 installfile() {
-  goipath="${1}"
-  file="${2}"
-  file="${file#./}"
-  [[ -d "${file}" && ! -L "${file}" ]] && srcdir="${file}" || srcdir=$(dirname "${file}")
-  destdir="${prefix}${gopath}/src/${goipath}/${srcdir}"
-  destdir="${destdir%/.}"
-  dir="${destdir}"
-  dirs=()
-  while [[ ! -e "${dir}" ]] ; do
-    dirs=("$dir" "${dirs[@]}")
-    dir=$(dirname "${dir}")
-  done
-  for dir in "${dirs[@]}" ; do
-    install -m 0755 -vd "${dir}"
-    if $(echo "${dir}" | grep -q "^${prefix}${gopath}/src/${goipath}") ; then
-      touch -r             ".${dir#${prefix}${gopath}/src/${goipath}}" "${dir}"
-    fi
-    echo "%dir \"${dir#${prefix}}\"" >> ${filelist}
-  done
-  if [[ -L "$file" ]] ; then
-    ln -s $(readlink "${file}") "${destdir}/$(basename ${file})"
-    touch -h      -r "${file}"  "${destdir}/$(basename ${file})"
+goipath="${1}"
+file="${2}"
+for proot in "$(realpath -sm ${PWD})" \
```

Comment:
I like the previous version more. I am getting lost here. Can you at least explain what the new code does in comments?